### PR TITLE
products page

### PR DESCRIPTION
**Pull request by MicLi01**

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "echo \"Error: no test specified\" && exit 1",
+    "startdev": "nodemon index.js"
```
Comment (synzen) [inline server/package.json]: Forgot a comma after here

File: server/routes/items.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const express = require('express')
+const router = express.Router();
+const itemHandler = require('../handlers/itemHandler');
+
+router.post('/create', itemHandler.createItem);
+router.post('/retrieve', itemHandler.retrieveItem);
```
Comment (synzen) [inline server/routes/items.js]: this should probably be a GET to stay consistent with retrieveall - if you do change it, be sure to remember to change this too

https://github.com/CSC59939/QuickBorrow/blob/5d15c049d756c9d025a0f905af4d9cabed0a1420/client/src/js/actions/index.js#L16

File: client/src/js/actions/index.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import axios from "axios";
+
+import { FETCH_PRODUCTS, FETCH_PRODUCT } from "./types";
+
+// Fetch all products
+export const fetchProducts = () => {
+  return async function(dispatch) {
+    const res = await axios.post("/server/items/retrieveall");
```
Comment (synzen) [inline client/src/js/actions/index.js]: this should be a GET according to your route definition

https://github.com/CSC59939/QuickBorrow/blob/5d15c049d756c9d025a0f905af4d9cabed0a1420/server/routes/items.js#L7

File: server/routes/items.js (hunk 2 above)
Comment (MicLi01) [reply server/routes/items.js]: modified /retrieve to be get request

**Comment by synzen** [comment]
btw the routes can also be modeled with REST principles -

`GET /api/items` -> replaces `/server/items/retrieveall`
`GET /api/items/:id` -> replaces `server/items/retrieve` that requires JSON body

but this is extra - whatever works 👍

**Comment by yliu0715** [comment]
@synzen Can you take a look at the App.js file? I'm not sure which lines we need and which lines we don't.

**Comment by synzen** [comment]
Conflicts have been fixed  https://github.com/CSC59939/QuickBorrow/pull/24/commits/be1f887dafd7c05a0e2bbcfcf1478d0bc2ef972b and https://github.com/CSC59939/QuickBorrow/pull/24/commits/bf81c4356eceb5700648e27fcee6445430596dce

**Pull request merged**